app/scripts/views/mixins/open-confirmation-email-mixin.js

```diff
@@ -0,0 +1,96 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+
+/**
+ * A view mixin that helps confirm* views show the "Open Gmail" button
+ * directly from FxA.
+ */
+define(function (require, exports, module) {
+  'use strict';
+
+  var BaseView = require('views/base');
+  var t = BaseView.t;
+
+  return {
+    events: {
+      'click #open-webmail': '_webmailTabOpened'
+    },
+
+    /**
+     * Get the webmail URL for the given email
+     *
+     * @param {string} email
+     * @returns {string}
+     */
+
+    _isGmail: function (email) {
+      return /@gmail\.com$/.test(email);
+    },
+
+    _isHotmail: function (email) {
+      return /@hotmail\.com$/.test(email);
+    },
+
+    _isYahoo: function (email) {
+      return /@yahoo\.com$/.test(email);
+    },
+
+    getWebmailLink: function (email) {
+      var linkToBeReturned;
+
+      if (this._isGmail(email)){
+        linkToBeReturned = 'https://mail.google.com/mail/u/?authuser=' + encodeURIComponent(email);
+      } else if (this._isHotmail(email)){
+        linkToBeReturned = 'https://hotmail.com';
+      } else if (this._isYahoo(email)){
+        linkToBeReturned = 'https://login.yahoo.com/config/login';
+      }
+
+      return linkToBeReturned;
+    },
+
+    /**
+     * Check if the `Open Webmail` button should be visible
+     *
+     * @param {string} email
+     * @returns {boolean}
+     */
+    isOpenWebmailButtonVisible: function (email) {
+      //The "Open webmail" button is only visible in certain contexts
+      // we do not show it in mobile context because it performs worse
+      return this.broker.hasCapability('openWebmailButtonVisible') &&
+            (this._isGmail(email) || this._isHotmail(email) || this._isYahoo(email));
+    },
+
+    getWebmailButtonText: function (email) {
+      var txtToBeReturned;
+
+      if (this._isGmail(email)){
+        txtToBeReturned = t('Open Gmail');
+      } else if (this._isHotmail(email)){
+        txtToBeReturned = t('Open Hotmail');
+      } else if (this._isYahoo(email)){
+        txtToBeReturned = t('Open Yahoo');
+      }
+
+      return txtToBeReturned;
+    },
+
+    _webmailTabOpened: function (event) {
+      var clicked = event.target.text;
+      var emailProvider;
+
+      if (clicked.includes('Gmail')){
+        emailProvider = 'openGmail';
+      } else if (clicked.includes('Hotmail')){
+        emailProvider = 'openHotmail';
+      } else if (clicked.includes('Yahoo')){
+        emailProvider = 'openYahoo';
+      }
+
+      this.logViewEvent(emailProvider.concat('_clicked'));
```

Comment:
@vladikoff, I see 2 options: get the email provider from the event (easier, but may stumble on undefined) or try to get it from email (guess I'll have to store provider somewhere in mixin, but then will have to be careful to reset it if user changes it (say she tries gmail then hotmail)...
So i opted to get info from DOM anyways. Better change it?